### PR TITLE
Remove FXIOS-5859 [v114] Get rid of SwiftyJSON from KeychainStore

### PR DESCRIPTION
**Pull request by mirscode**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-5859)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/13377)

### Description
Removes SwiftyJSON from Keychain Store.

The ticket mentioned adding Unit Tests if necessary, and I can certainly test out the functions in this class if we still want this. Please let me know.

### CODE REVIEW

**Comment by mte-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Edited 1 files</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Created 0 files</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against d13a2155730d25fad3fc6c3369668c7823fb7f88
</p>

**Comment by OrlaM** [comment]
@mirscode Thanks for the contribution. It would be great if you could write some tests. If the tests pass both before your change and after then it would add a lot of confidence that the change doesn't introduce any issues.

**Comment by mirscode** [comment]
> @mirscode Thanks for the contribution. It would be great if you could write some tests. If the tests pass both before your change and after then it would add a lot of confidence that the change doesn't introduce any issues.

Sounds good, I'll add the tests as soon as I can :)

**Comment by mirscode** [comment]
@OrlaM I wrote unit tests for `KeychainStore` and they pass (without any edits) before/after my changes. Please LMK if this works.

File: Shared/KeychainStore.swift [hunk 1]
```diff
@@ -3,27 +3,19 @@
...
 
 public class KeychainStore {
     public static let shared = KeychainStore()
 
     private let keychainWrapper: MZKeychainWrapper
 
-    private init() {
+    public init() {
         self.keychainWrapper = MZKeychainWrapper.sharedClientAppContainerKeychain
     }
```
[the remark this answers is not in the document]
Comment (OrlaM) [reply Shared/KeychainStore.swift:15]: That is a pattern we are trying to adopt as we slowly modernize and add more tests to the codebase so if you pick up any future tickets then feel free to make that kind of change as needed.
Thanks for adding tests, I will merge once CI passes.

**Review by OrlaM — APPROVED** [review]

**Comment by OrlaM** [comment]
It looks like there are some swiftlint warnings that need to be fixed.

**Comment by mirscode** [comment]
> It looks like there are some swiftlint warnings that need to be fixed.

Ah, sorry about that. I put up another commit appeasing swiftlint and I also ran it locally to ensure zero violations. Hoping it will pass smoothly this time around. Thanks!

**Comment by lmarceau** [comment]
@mirscode it seems two tests failed on Bitrise 👀
Screen Shot 2023-04-06 at 9 19 00 AM

**Comment by mirscode** [comment]
Since `MZKeychainWrapper` is a package and its functions are marked as `open`, I felt it would be easiest to override those functions by creating a `FakeKeychainWrapper` class that inherits from KeychainWrapper, and uses a `fakeDictionary` of type `[String: Any]`. This way, we can test out the functionality of `KeychainStore` without using real `MZKeychainWrapper` functionality when our test suite is running.

I believe this is the best path forward right now given that there isn't much to go off of in the logs and calling `MZKeychainWrapper.wipeKeychain()` doesn't seem like the right call for testing `KeychainStore`.

Please take a look at the latest commit and let me know your thoughts. If this seems acceptable, then let's create an issue to implement all overridable functions from `MZKeychainWrapper` into `FakeKeychainWrapper`.

If this isn't acceptable, we can take three routes:
1. Remove the latest commit and attempt to run the test suite after calling `MZKeychainWrapper.wipeKeychain()` to check if that will work
2. Open a PR against the package housing `MZKeychainWrapper` to make `MZKeychainWrapper` a protocol so we can create a Fake
3. Remove tests that require `MZKeychainWrapper`

**Comment by OrlaM** [comment]
Thanks for the update! It's all green now on the CI, ignore the one failure it's actually green but we have an issue that occurs every so often with syncing the state to github.

**Pull request merged**